tests/ballerina-test/src/test/resources/test-src/net/http/resiliency/failover-connector-test.bal

```diff
@@ -0,0 +1,141 @@
+import ballerina.net.http;
+import ballerina.net.http.resiliency;
+import ballerina.mime;
+
+http:Retry retryConfiguration = {count:0};
+
+http:HttpClient endPoint1 = (http:HttpClient) create MockHttpFailOverClient("http://invalidEP", {});
+http:HttpClient endPoint2 = (http:HttpClient) create MockHttpFailOverClient("http://localhost:8080",
+                                                   {endpointTimeout:5000,
+                                                       keepAlive:true,
+                                                       retryConfig:retryConfiguration});
+
+http:HttpClient[] failoverGroup = [endPoint1, endPoint2];
+int[] errorCodes = [404, 502];
+
+resiliency:FailoverConfig errorCode = {failoverCodes:errorCodes};
+
+function testSuccessScenario () (http:InResponse, http:HttpConnectorError) {
+    endpoint<http:HttpClient> ep {
+        create resiliency:Failover(failoverGroup, errorCode);
+    }
+
+    http:InResponse clientResponse = {};
+    http:HttpConnectorError err;
+
+    mime:MediaType contentType = mime:getMediaType("application/xml");
+    mime:Entity requestEntity = {};
+    requestEntity.xmlData = xml `<name>Ballerina</name>`;
+    requestEntity.contentType = contentType;
+
+    http:OutRequest outReq = {};
+    outReq.setEntity(requestEntity);
+
+    int counter = 0;
+
+    while (counter < 2) {
+        clientResponse, err = ep.post("/", outReq);
+        counter = counter + 1;
+    }
+
+    return clientResponse, err;
+}
+
+function testFailureScenario () (http:InResponse, http:HttpConnectorError) {
+    endpoint<http:HttpClient> ep {
+        create resiliency:Failover(failoverGroup, errorCode);
+    }
+
+    http:InResponse clientResponse = {};
+    http:HttpConnectorError err;
+
+    mime:MediaType contentType = mime:getMediaType("application/xml");
+    mime:Entity requestEntity = {};
+    requestEntity.xmlData = xml `<name>Ballerina</name>`;
```

Comment:
This way of setting xml is no longer valid. Please use the latest API. eg:  requestEntity.setXml(..).